Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BaseShapeComponentMode.cpp

```diff
@@ -0,0 +1,368 @@
+    }
+
+    void BaseShapeComponentMode::SetupCluster()
+    {
+        ViewportUi::ViewportUiRequestBus::EventResult(
+            m_clusterId,
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::CreateCluster,
+            ViewportUi::Alignment::TopLeft);
+
+        m_buttonIds[static_cast<AZ::u32>(ShapeComponentModeRequests::SubMode::Dimensions)] =
+            RegisterClusterButton(ViewportUi::DefaultViewportId, m_clusterId, "Scale", DimensionsTooltip);
+        m_buttonIds[static_cast<AZ::u32>(ShapeComponentModeRequests::SubMode::TranslationOffset)] =
+            RegisterClusterButton(ViewportUi::DefaultViewportId, m_clusterId, "Move", TranslationOffsetTooltip);
+
+        const auto onButtonClicked = [this](ViewportUi::ButtonId buttonId)
+        {
+            if (buttonId == m_buttonIds[static_cast<AZ::u32>(ShapeComponentModeRequests::SubMode::Dimensions)])
+            {
+                SetShapeSubMode(ShapeComponentModeRequests::SubMode::Dimensions);
+            }
+            else if (buttonId == m_buttonIds[static_cast<AZ::u32>(ShapeComponentModeRequests::SubMode::TranslationOffset)])
+            {
+                SetShapeSubMode(ShapeComponentModeRequests::SubMode::TranslationOffset);
+            }
+        };
+
+        m_modeSelectionHandler = AZ::Event<ViewportUi::ButtonId>::Handler(onButtonClicked);
+        ViewportUi::ViewportUiRequestBus::Event(
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::RegisterClusterEventHandler,
+            m_clusterId,
+            m_modeSelectionHandler);
+    }
+
+    ShapeComponentModeRequests::SubMode BaseShapeComponentMode::GetShapeSubMode() const
+    {
+        return m_subMode;
+    }
+
+    void BaseShapeComponentMode::SetShapeSubMode(ShapeComponentModeRequests::SubMode mode)
+    {
+        AZ_Assert(mode < ShapeComponentModeRequests::SubMode::NumModes, "Submode not found:%d", static_cast<AZ::u32>(mode));
+        m_subModes[static_cast<AZ::u32>(m_subMode)]->Teardown();
+        const auto modeIndex = static_cast<AZ::u32>(mode);
+        AZ_Assert(modeIndex < m_buttonIds.size(), "Invalid mode index %i.", modeIndex);
+        m_subMode = mode;
+        m_subModes[modeIndex]->Setup(g_mainManipulatorManagerId);
+        m_subModes[modeIndex]->AddEntityComponentIdPair(m_entityComponentIdPair);
+
+        ViewportUi::ViewportUiRequestBus::Event(
+            ViewportUi::DefaultViewportId, &ViewportUi::ViewportUiRequestBus::Events::ClearClusterActiveButton, m_clusterId);
+
+        ViewportUi::ViewportUiRequestBus::Event(
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::SetClusterActiveButton,
+            m_clusterId,
+            m_buttonIds[modeIndex]);
+    }
+
+    void BaseShapeComponentMode::ResetShapeSubMode()
+    {
+        m_subModes[static_cast<AZ::u32>(m_subMode)]->ResetValues();
```

Comment:
nit/suggestion: Could do this to avoid the cast twice

```
auto& subMode = m_subModes[static_cast<AZ::u32>(m_subMode)];
subMode->ResetValues();
subMode->UpdateManipulators();
```